suitcase/fields.py

```diff
@@ -656,6 +725,51 @@ def setval(self, value):
         return self._get_parent_field().setval(value)
 
 
+class FieldArray(BaseField):
+    """Field which contains a list of some other field.
+
+    In some protocols, there exist repeated substructures which are present in a
+    variable number. The variable nature of these fields make a DispatchField
+    and DispatchTarget combination unsuitable; instead a FieldArray may be
+    used to pack/unpack these fields to/from a list.
+

```

Comment:
Since `TypeField` and `FieldArray` are a little tricky, I'm wondering if we can include a small example in the docstring.  I find that this is extremely valuable when reading documentation.  Here's what I came up with for `FieldArray`.

```python
class PascalString16(Structure):
    length = LengthField(UBInt16())
    value = Payload(length)

class FamousQuoteRecord(Structure):
    author = PascalString16()
    quotes = FieldArray(PascalString16)  # variable number of separate quotes
```